Zend/zend_compile.h

```diff
@@ -257,13 +257,16 @@ typedef struct _zend_oparray_context {
 /* disable inline caching */
 #define ZEND_ACC_NEVER_CACHE          0x400000
 
-#define ZEND_ACC_VARIADIC				0x1000000
+#define ZEND_ACC_VARIADIC			0x1000000
 
 #define ZEND_ACC_RETURN_REFERENCE		0x4000000
 #define ZEND_ACC_DONE_PASS_TWO			0x8000000
 
 /* class has magic methods __get/__set/__unset/__isset that use guards */
-#define ZEND_ACC_USE_GUARDS				0x1000000
+#define ZEND_ACC_USE_GUARDS			0x1000000
+
+/* class may be weakly referenced */
+#define ZEND_ACC_USE_WEAK_REFS			0x2000000
```

Comment:
What's the motivation behind this flag? Right now you're using this to exclude weak-refs to weak-refs. That doesn't seem like a terribly useful pattern, but is there a technical reason why we need to prohibit it?

Reply:
That's just for internal classes to exclude themselves from being weakly referenced, for the case where the internal class relies in some way on the reference count being accurate (as horrible as that seems).

I excluded weakrefs pending discussion about weakly referenced weakrefs and what that should mean, and whether that should be implemented as a weakmap, or something else ...

Reply:
Do you have an example of an internal class that would not be able to support weak referencing?

Reply:
Off the top of my head, no concrete example, but I noted that in python it's an opt-in feature for internal classes ... and it would seem like a reasonable thing to be able to switch off. 

Happy to remove it since I can't come up with anything concrete ...